propolis/src/vmm/machine.rs

```diff
@@ -446,74 +407,34 @@ impl Builder {
         }
     }
 
-    fn last_sysmem_addr(&self) -> Result<usize> {
-        let last_mem_seg = self
-            .memmap
-            .iter()
-            .filter(|(_addr, _len, (ent, _name))| {
-                matches!(ent, MapKind::SysMem(_, _))
-            })
-            .last();
-        if let Some((start, len, (_, _))) = last_mem_seg {
-            Ok(start + len)
-        } else {
-            Err(Error::new(ErrorKind::InvalidData, "missing guest memory"))
-        }
-    }
-
     fn prep_mem_map(&self, hdl: &VmmHdl) -> Result<ASpace<MapEnt>> {
-        let last_sysmem = self.last_sysmem_addr()?;
```

Comment:
So, I'm not sure I fully understand this previous mechanism of "calculate guard space based on last sysmem address".

- Doesn't this calculation ignore ROM?
- Isn't it implicitly dependent on the order of SysMem registration?

The new version just sums the memory, and adds space for guard pages.

Reply:
Followed-up offline, using a new allocator within GuardSpace for this now, and calculating the "highest seen guest address" as the total size of that region.